discord/ext/ipc/client.py

```diff
@@ -103,57 +110,98 @@ async def init_sock(self):
         )
         log.info("Client connected to %s", self.url)
 
+        if self._worker is None:
+            self._worker = self.loop.create_task(self._receive_requests())
+        if self._sender is None:
+            self._sender = self.loop.create_task(self._send_requests())
+
         return self.websocket
 
-    async def request(self, endpoint: str, **kwargs):
+    async def request(self, endpoint: str, timeout: float = None, **kwargs):
         """Make a request to the IPC server process.
 
         Parameters
         ----------
         endpoint: str
             The endpoint to request on the server
+        timeout: float
+            The number of seconds to wait for a response from the server
+            before timing out and raising :exc:`asyncio.TimeoutError`.
+            By default this is ``None`` and waits forever.
         **kwargs
             The data to send to the endpoint
+
+        Raises
+        -------
+        asyncio.TimeoutError
+            If a timeout is provided and it was reached.
         """
         log.info("Requesting IPC Server for %r with %r", endpoint, kwargs)
         if not self.session:
             await self.init_sock()
 
+        # generate a unique identifier for this request
+        # there is probably a better way to do this
+        nonce = uuid.uuid1()
```

Comment:
```suggestion
        nonce = os.urandom(16).hex()
```

Personally i'd use something like this.

```py
In [1]: %timeit uuid.uuid1()
1.98 µs ± 9.62 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [2]: %timeit os.urandom(16).hex()
130 ns ± 0.409 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
```

You'll need to change the imports though.